ethexe/runtime/common/src/journal.rs

```diff
@@ -66,18 +67,20 @@ impl<S: Storage> Handler<'_, S> {
             return;
         }
 
+        let dispatch_origin = self.dispatch_origin;
+
         self.update_state(dispatch.source(), |state, storage, transitions| {
             if let Ok(non_zero_delay) = delay.try_into() {
                 let expiry = transitions.schedule_task(
                     non_zero_delay,
                     ScheduledTask::SendUserMessage {
                         message_id: dispatch.id(),
-                        to_mailbox: dispatch.source(),
+                        to_mailbox: (dispatch.source(), dispatch_origin),
                     },
                 );
 
                 let user_id = dispatch.destination();
-                let dispatch = Dispatch::from_stored(storage, dispatch);
+                let dispatch = Dispatch::from_stored(storage, dispatch, dispatch_origin);
```

Comment:
rename fn into `from_core_stored` pls to avoid misunderstanding

Reply:
Fixed